Clarify the object constructor curly braces (#1496)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-types.adoc

  (y)
}
----

> Notice the expression between the _parentheses_ (such as `(x)` where `x` is the variable shown in the header) should return a single object or an array of objects. In this example, `(x)` returns and array of objects, and `(y)` returns an object with just one key/value. All of the key/values from each object in that array get flattened "in place" by the object constructor curly braces into the outer object. The object constructor curly braces also operate on the `(y)` expression to flatten the single key/value "in place" from that object into the outer object. So the output looks like this:

Reviewer: ```suggestion
Notice the expression between the _parentheses_ (such as `(x)` where `x` is the variable shown in the header) returns a single object or an array of objects. In this example, `(x)` returns an array of objects, and `(y)` returns an object with just one key-value pair. The object constructor curly braces flattens all of the key-value pairs from each object "in place," into the outer object. The object constructor curly braces also operate on the `(y)` expression to flatten the single key-value pair from that object into the outer object. The output looks like this:
```

Author: Notice the expression between the _ evaluation parentheses_ (such as `(x)` where `x` is the variable shown in the header) must evaluate to something with key/value pairs that can be extracted by the _object constructor curly braces_, either an object such as `{ "a":"one", "b":"two"}`, or an array of objects such as `[{"a":"one"},{"b":"two"}]`. In this example, `(x)` returns an array of objects, and `(y)` returns an object. In this example the object constructor curly braces `{ }` are applied to a comma separated list of DataWeave types.  The first element in the list is a key/value pair `"a":"a"` so it is left alone, in place, as the first element of the newly constructed object. When applied to the second element (x), the object constructor curly braces flattens all of the child key-value pairs from the evaluated Array of Objects `(x)` "in place" into the resulting object. In this case, it first extracts  the key/value `"b":"b"` followed by extracting the two key/values from `"c":"c", "d":"d"`. Next, when applied to the third element `(y)` in the comma separated list, the object constructor curly braces extracts each key/value from the evaluated Object `(y)` "in place" to flatten the single key-value pair from that object into the outer object. In this case it is just the single key/value pair `"e":"e"`. The output looks like this:

Reviewer: Wondering if something like this would work:

The _evaluation parentheses_ around the variables (`(x)` and `(y)`) in the body of the script above enable the object constructor curly braces (`{ }`) surrounding the body of the script to extract the key-value pairs from the values of `x` and `y` and convert those values into a collection of key-value pairs within an object. Without the parentheses (for example, if you use `x` instead of `(x)`), the script returns an error `Invalid input '}', expected Namespace or Attribute&lt;'@('(Name:Value)+')'&gt; `.

The object constructor curly braces require a data construct that contains key-value pairs, specifically an object, such as `{ "a":"one", "b":"two"}`, or an array of objects, such as `[{"a":"one"},{"b":"two"}]` so that it can produce a valid object. Because the first element within the object constructor curly braces is the key-value pair `a: "a"`, the object constructor curly braces leaves it as the first element in the output object, without any modifications. However, the object constructor curly braces extracts the child elements of the `(x)` and `(y)` values from the containing array or object and lists them in the output object.

}
----

> The _evaluation parentheses_ around the variables in the body of the script above (`(x)` and `(y)`) enable the object constructor curly braces (`{ }`) surrounding the body of the script to extract the key-value pairs from the values of `x` and `y` and convert those values into a collection of key-value pairs within an object. Without the parentheses (for example, if you use `x` instead of `(x)`), the script returns an error.

Reviewer 2: the script **above** > the **previous** script

This first sentence is a bit too long, I'm not sure how to reword it, but if it is possible to do so that would be good.

> The _evaluation parentheses_ around the variables in the body of the script above (`(x)` and `(y)`) enable the object constructor curly braces (`{ }`) surrounding the body of the script to extract the key-value pairs from the values of `x` and `y` and convert those values into a collection of key-value pairs within an object. Without the parentheses (for example, if you use `x` instead of `(x)`), the script returns an error `Invalid input '}', expected Namespace or Attribute&lt;'@('(Name:Value)+')'&gt;`.

Reviewer 3: I would just delete the error message because that could change over time. Like this: `Without the parentheses (for example, if you use `x` instead of `(x)`), the script returns an error.`
The rest looks correct for me.

Reviewer: done.
